src/mlpack/methods/ann/layer/radial_basis_function.hpp

```diff
@@ -0,0 +1,174 @@
+/**
+ * @file radial_basis_function.hpp
+ * @author Himanshu Pathak
+ *
+ * Definition of the Radial Basis Function module class.
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RBF_HPP
+#define MLPACK_METHODS_ANN_LAYER_RBF_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/regularizer/no_regularizer.hpp>
+
+#include "layer_types.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+
+/**
+/**
+ * Implementation of the Radial Basis Function layer. The RBF class when use with a 
+ * non-linear activation function acts as a Radial Basis Function which can be used
+ * with Feed-Forward neural network.
+ *
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat,
+    typename RegularizerType = NoRegularizer
```

Comment:
why do we need regulariser?

Reply:
I will remove it in next commit added because of previous implementation.

Reply:
Can you add a template parameter for the activation function? That is a part of the radial basis function.